src/parallel_simulation/network/link.rs

```diff
@@ -1,36 +1,36 @@
 use crate::io::network::IOLink;
 use crate::parallel_simulation::network::flowcap::Flowcap;
-use crate::parallel_simulation::vehicles::Vehicle;
 use std::collections::VecDeque;
+use std::fmt::Debug;
 
 #[derive(Debug)]
-pub enum Link {
-    LocalLink(LocalLink),
-    SplitInLink(SplitInLink),
+pub enum Link<V: Debug> {
```

Comment:
All the stuff in the `network` package is used by `parallel_simulation` as well as `mpi::simulation`. Both use different implementations of `vehicle`, `agent`, and so on. I made this generic, so we can use the same network with different entity types for both implementations. 

Depending what we decide to do when "tidying up" the repo, this generic could be removed.  

Actually, we should probably enforce a bound on `NodeVehicle` since that is the trait which is required when moving vehicles from one `Node` to the next.